.github/workflows/scripts/slack_gem_notifications/notifications_methods.rb

```diff
@@ -0,0 +1,55 @@
+# encoding: utf-8
+# This file is distributed under New Relic's license terms.
+# See https://github.com/newrelic/newrelic-ruby-agent/blob/main/LICENSE for complete details.
+
+require 'time'
+require "httparty"
+
+def check_for_updates(watched_gems)
+  return if gem_list_empty?(watched_gems)
+  watched_gems.each do |gem_name|
+    return unless is_valid_gem?(gem_name)
+    versions = get_gem_info(gem_name)
+    send_bot(gem_name, versions) if gem_updated?(versions)
+  end
+end
+
+def gem_list_empty?(watched_gems)
+  if watched_gems.empty?
+    raise 'Nothing to see here! watched_gems cannot be empty'
+    exit
+  end
+
+  watched_gems.empty?
+end
+
+def is_valid_gem?(gem_name)
+  res = HTTParty.get("https://rubygems.org/api/v1/versions/#{gem_name}.json")
+
+  res.code == 200
```

Comment:
`res` here is an instance of the class `HTTParty::Response`. Does that class offer us any way to confirm that the response was successful apart from us having to manually check the numeric HTTP response code?

Reply:
sure enough I can do `res.success?` ✨ nice call out, thank you